src/fl/federation/client.py

```diff
@@ -47,32 +47,32 @@ def fit(self, parameters, config):
         trainer.fit(self.model, datamodule=self.data_module)
         return self.get_parameters(), self.data_module.train_len(), {}
 
-    def calculate_r2_score(self, trainer: Trainer, loader: DataLoader) -> float:
+    def calculate_loader_metrics(self, trainer: Trainer, loader: DataLoader) -> Tuple[float, float]:
         preds = trainer.predict(self.model, loader)
-        preds = torch.cat(preds, dim=0)
-        y_true = torch.cat([batch[1] for batch in iter(loader)])
-        r2 = r2_score(y_true.detach().cpu().numpy(), preds.detach().cpu().numpy())
-        return r2
-
-    def calculate_train_val_r2(self, trainer: Trainer) -> float:
-        train_loader, val_loader = self.data_module.predict_dataloader()
+        preds = torch.cat(preds, dim=0).detach().cpu().numpy()
+        y_true = torch.cat([batch[1] for batch in iter(loader)]).detach().cpu().numpy()
         
-        return self.calculate_r2_score(trainer, train_loader), self.calculate_r2_score(trainer, val_loader)
+        mse = mean_squared_error(y_true, preds)
```

Comment:
Please don't forget that in the future we will continue binary phenotypes as well so MSE won't be relevant there. Maybe we put the list of metrics in the arguments? Maybe as a dict: metrics_dict={'r2': r2_score, 'MSE': mean_squared_error} and in the body:
metrics_values_dict = {k: v(y_true, preds) for k, v in metrics_dict.items()}